.flake8

```
@@ -0,0 +1,20 @@
+# Can't yet be moved to the pyproject.toml due to https://gitlab.com/pycqa/flake8/-/issues/428#note_251982786
+[flake8]
+max-line-length = 88
+# switched off since they conflict with black's standards
+ignore = F401,  # module imported but unused -> required for Scanpys API
+         W503,  # line break before a binary operator -> black does not adhere to PEP8
+         W504,  # line break occured after a binary operator -> black does not adhere to PEP8
+         E501,  # line too long -> we accept long comment lines; black gets rid of long code lines
+         E203,  # whitespace before : -> black does not adhere to PEP8
+         E231,  # missing whitespace after ,', ';', or ':' -> black does not adhere to PEP8
+         E402,  # module level import not at top of file -> required to circumvent circular imports for Scanpys API
+         E126,  # continuation line over-indented for hanging indent -> black does not adhere to PEP8
+         E266,  # E266 too many leading '#' for block comment -> Scanpy allows them for comments into sections
+         E731   # Do not assign a lambda expression, use a def -> Scanpy allows lambda expression assignments
+exclude =
+    .git,
+    __pycache__,
+    build,
+    docs/_build
+    dist
```

```suggestion
    docs/_build,
    dist,
per-file-ignores = 
    # F811 Redefinition of unused name from line, does not play nice with pytest fixtures
    tests/test*.py: F811,
```

Supposedly, you're not allowed to have comments on the same lines as entries (even though this clearly works above). See: https://flake8.pycqa.org/en/latest/user/configuration.html